### DISCUSSION TITLE
Only update property on "keydown.enter"

### DISCUSSION THREAD

**Discussion by oliverbj** [livewire/livewire#1086] [4. General]
Hi there!

I have a simple input field, where my users can define "labels" (for example blog post labels). Imagine an input field, where for each string you type and press enter, it will add that string as a tag below the input field.

This is my implementation:

```php
//Livewire component:
class MyComponent extends Component
{
   public $labels = [];
}
//Livewire component (blade view):
<input wire:model.lazy="labels.name" x-on:keydown.enter="document.getElementById('labels').value=''" id="labels">
@foreach($labels as $label)
  {{$label}}
@endforeach
```

However the above does not work as I expect. When pressing enter, the Alpine.js code is executed and clearing the input field, but nothing is added to `$labels`. I can only add to `$labels` if I click somewhere on the page.

Any ideas?

**Comment by oliverbj** [comment]
I have solved this by doing the following:

```html
<input  id="labels" type="text" placeholder="Add labels by pressing enter..."
                            wire:keydown.enter="addLabel($event.target.value)"
                            x-on:keydown.enter="document.getElementById('labels').value=''">
```

Then in my component class:

```php
public $labels = [];

public function addLabel(string $name)
{
    if (!in_array($name, $this->labels) && !empty($name)) {
        array_push($this->labels, $name);
    }
}
```

**Reply by mityukov** [reply]
> wire:keydown.enter="addLabel($event.target.value)"

When I do something like this, I get "TypeError: Cannot read property 'target' of undefined". But when it's wire:change — there is no such error…

```
wire:keydown.enter="updateEmail({{ $loop->index }}, $event.target.value)" // has error
wire:change="updateEmail({{ $loop->index }}, $event.target.value)" // has NO error

```

**Reply by mityukov** [reply]
I was wrong. The issue is connected to `wire:loading.attr="disabled"` directive…
